internal/uploader/events_uploader_test.go

```diff
@@ -392,23 +393,59 @@ var _ = Describe("UploadEvents", func() {
 		Expect(err).ShouldNot(HaveOccurred())
 	})
 	It("fails to upload event data when there's no data", func() {
-		createOCMPullSecret(*mockK8sClient, true)
+		createOCMPullSecretWithToken(*mockK8sClient)
 		mockEvents.EXPECT().V2GetEvents(ctx, common.GetDefaultV2GetEventsParams(nil, nil, nil, models.EventCategoryMetrics, models.EventCategoryUser)).
 			Return(
 				nil, errors.New("no events found")).
 			Times(1)
 		err := uploader.UploadEvents(ctx, &common.Cluster{PullSecret: ""}, mockEvents)
 		Expect(err).To(HaveOccurred())
 	})
-	It("fails to uploads event data when headers can't be set", func() {
-		createOCMPullSecret(*mockK8sClient, false)
+	It("fails to uploads event data when malformed pullsecret", func() {
```

Comment:
Headers depend on pullsecret, but we validate it beforehand, so it seems that is not possible to have invalid/empty pullsecret and fail on headers, it would fail on pullsecret check